### PR TITLE
[ColorPicker] Fix errant text selection

### LINKED ISSUE #1542: Color picker selects text outside of picker while dragging in Safari
## Issue summary

When dragging the color picker in Safari text elements outside of the color picker container are highlighted. 

## Expected behavior

When dragging the color picker, text elements should not be highlighted. 

## Actual behavior

screencast 2019-05-21 11-16-05

## Steps to reproduce the problem

1. Add ColorPicker component and adjacent text element to project
2. Open project in Safari
3. Click and drag the color picker and bring mouse outside of the picker to hover over text

## Reduced test case

https://codesandbox.io/s/shopify-polaris-react-typescript-playground-sqfcn?fontsize=14

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 3.15
- Browser: Safari Version 12.0.3 (14606.4.5)
- Device: Macbook Pro
- Operating System: MacOS 10.14.3

### PR DESCRIPTION
**Pull request by alexieyizhe**
### WHY are these changes introduced?

Fixes #1542. Text outside the color picker was being selected when dragging the picker.

### WHAT is this pull request doing?

No UI changes. Simply prevented the mousedown event from bubbling up to other elements in the DOM to stop other text from being able to detect they were being dragged over to select.

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing) (iPhone XS, Safari/Chrome)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers) (MBP, Safari/Chrome)
* [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [x] ~Updated the component's `README.md` with documentation changes~ N/A
* [x] ~[Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide~ N/A

### CODE REVIEW

File: UNRELEASED.md [hunk 1]
```diff
@@ -16,6 +16,8 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 
 ### Bug fixes
 
+- Fixed text and other elements from being selected in Safari when dragging the color picker. ([#1562](https://github.com/Shopify/polaris-react/pull/1562))
```
Comment (tmlayton) [inline UNRELEASED.md]:
```suggestion
- Fixed text and other elements from being selected in Safari when dragging the color picker ([#1562](https://github.com/Shopify/polaris-react/pull/1562))
```

File: src/components/ColorPicker/ColorPicker.tsx [hunk 2]
```diff
@@ -120,4 +125,11 @@ export default class ColorPicker extends React.PureComponent<Props, State> {
 
     onChange({hue, saturation, brightness, alpha});
   };
+
+  private handlePickerDrag = (
+    evnt: React.MouseEvent<HTMLDivElement, MouseEvent>,
```
Comment (tmlayton) [inline src/components/ColorPicker/ColorPicker.tsx]: For code consistency:

```suggestion
    event: React.MouseEvent<HTMLDivElement, MouseEvent>,
```

File: src/components/ColorPicker/ColorPicker.tsx [hunk 3]
```diff
@@ -120,4 +125,11 @@ export default class ColorPicker extends React.PureComponent<Props, State> {
...
+  ) => {
+    // prevents external elements from being selected
+    evnt.preventDefault();
```
Comment (tmlayton) [inline src/components/ColorPicker/ColorPicker.tsx]:
```suggestion
    event.preventDefault();
```

File: src/components/ColorPicker/ColorPicker.tsx [hunk 4]
```diff
@@ -70,7 +70,12 @@ export default class ColorPicker extends React.PureComponent<Props, State> {
     ) : null;
 
     return (
-      <div className={styles.ColorPicker} id={id}>
+      <div
+        className={styles.ColorPicker}
+        id={id}
+        onMouseDown={this.handlePickerDrag}
+        onMouseMove={this.handlePickerDrag}
```
Comment (tmlayton) [inline src/components/ColorPicker/ColorPicker.tsx]: Do we need the `mousemove` event since we are preventing the default for `mousedown`?

Comment (alexieyizhe) [reply src/components/ColorPicker/ColorPicker.tsx]: Good point! I don't think it's needed, but I will confirm

Comment (alexieyizhe) [reply src/components/ColorPicker/ColorPicker.tsx]: Doesn't seem to be affecting it, removed

**Comment by tmlayton** [comment]
Thank you so much for the contribution! 💜

**Comment by alexieyizhe** [comment]
Thanks for the feedback :) it was a a great learning experience

**Pull request merged**